### PR TITLE
Remove setEvaluateNodeInRegPair in Z CodeGenerator

### PR DESCRIPTION
**Pull request by shivammittal99**
Remove evaluateNodeInRegPair and fold as it returns false.

Closes: #2060 

Signed-off-by: Shivam Mittal <shivammittal99@gmail.com>

### CODE REVIEW

File: compiler/z/codegen/UnaryEvaluator.cpp [hunk 1]
```diff
@@ -77,7 +77,7 @@ extern void PRINT_ME(char * string, TR::Node * node, TR::CodeGenerator * cg);
 TR::Register *
 lconstHelper(TR::Node * node, TR::CodeGenerator * cg)
    {
-   TR_ASSERT( TR::Compiler->target.is32Bit() || cg->evaluateNodeInRegPair(node), "lconstHelper() is for 32bit only!");
+   TR_ASSERT( TR::Compiler->target.is32Bit(), "lconstHelper() is for 32bit only!");
```
Comment (fjeremic) [inline compiler/z/codegen/UnaryEvaluator.cpp]: May as well properly format this while we're in the area (extra space after `TR_ASSERT`).

**Comment by 0xdaryl** [comment]
@genie-omr build zlinux zos

File: compiler/z/codegen/OMRCodeGenerator.hpp [hunk 2]
```diff
@@ -600,32 +600,6 @@ class OMR_EXTENSIBLE CodeGenerator : public OMR::CodeGenerator
...
-
-   bool evaluateNodeInRegPair(TR::Node * node)
-   {
-   CS2::HashIndex hashIndex = 0;
-   if (!_nodesToBeEvaluatedInRegPairs.Locate(node->getGlobalIndex(), hashIndex))
-      return false;
-   else
-      return _nodesToBeEvaluatedInRegPairs.DataAt(hashIndex);
-   }
-
```
Comment (0xdaryl) [inline compiler/z/codegen/OMRCodeGenerator.hpp]: Please ensure the copyright header of this file updated to 2018.

**Comment by 0xdaryl** [comment]
Thanks for the changes.

For completeness, @genie-omr build zlinux zos

Normally I'd suggest you squash the commits together.  However, I've been itching to try out the "squash and merge" button in GitHub (and this PR is the perfect candidate) so I'll do the squashing and see what happens.  :-)

**Comment by 0xdaryl** [comment]
This PR affects Z files only.  There is no need to wait for the stalled macOS build in order to merge this.  Merging.

**Pull request merged**